src/app/_helpers/calcprogress.ts

```diff
@@ -0,0 +1,13 @@
+import { WorkflowExecution } from '../_models';
+
+export function calcProgress(ongoing: WorkflowExecution): number {
+  if (!ongoing.currentPluginIndex) {
+    return 0;
+  }
+
+  const pct: number = (ongoing.currentPluginIndex / ongoing.metisPlugins.length) * 100;
```

Comment:
you can leave out the `: number` type annotations. but it's a matter of style, this is also fine.